core/src/main/java/org/apache/calcite/plan/Strong.java

```diff
@@ -151,12 +151,41 @@ public static boolean allStrong(List<RexNode> operands) {
 
   /** Returns whether an expression is definitely not true. */
   public boolean isNotTrue(RexNode node) {
+    return isPredicateNotTrue(node) || isNull(node);
```

Comment:
I'm still not sure why we cannot merge `isNotTrue` and `isPredicateNotTrue` like below:

```
  public boolean isNotTrue(RexNode node) {
    switch (node.getKind()) {
    //TODO Enrich with more possible cases?
    case IS_NOT_NULL:
      return isNull(((RexCall) node).getOperands().get(0));
    case OR:
      return allNotTrue(((RexCall) node).getOperands());
    case AND:
      return anyNotTrue(((RexCall) node).getOperands());
    default:
      return isNull(node);
    }
  }

  /** Returns whether all expressions in a list are definitely not true. */
  private boolean allNotTrue(List<RexNode> operands) {
    for (RexNode operand : operands) {
      if (!isNotTrue(operand)) {
        return false;
      }
    }
    return true;
  }

  /** Returns whether any expressions in a list are definitely not true. */
  private boolean anyNotTrue(List<RexNode> operands) {
    for (RexNode operand : operands) {
      if (isNotTrue(operand)) {
        return true;
      }
    }
    return false;
  }
```

Reply:
it is doable, it covers more cases this way hence the new change in RelOptRules.xml
The case is AND/OR(DEFINITELY FALSE, DEFINITELY NULL) -> DEFINITELY FALSE
done!